op/opstatus.go

```diff
@@ -14,59 +20,71 @@ func NewStatus(name, description string) *Status {
 
 // AddOwner adds an owner entry. Each can have a name, a slack channel or both.
 // Multiple owner entries are allowed.
-func (os *Status) AddOwner(name, slack string) *Status {
-	os.owners = append(os.owners, owner{name: name, slack: slack})
-	return os
+func (s *Status) AddOwner(name, slack string) *Status {
+	s.owners = append(s.owners, owner{name: name, slack: slack})
+	return s
 }
 
 // AddLink adds a URL link. Multiple are allowed and each should have a brief
 // description.
-func (os *Status) AddLink(description, url string) *Status {
-	os.links = append(os.links, link{description: description, url: url})
-	return os
+func (s *Status) AddLink(description, url string) *Status {
+	s.links = append(s.links, link{description: description, url: url})
+	return s
 }
 
 // SetRevision sets the source control revision string, typically a git hash.
-func (os *Status) SetRevision(revision string) *Status {
-	os.revision = revision
-	return os
+func (s *Status) SetRevision(revision string) *Status {
+	s.revision = revision
+	return s
 }
 
 // AddChecker adds a function that can check the applications health.
 // Multiple checkers are allowed.  The checker functions should be capable of
 // being called concurrently (with each other and with themselves).
-func (os *Status) AddChecker(name string, checkerFunc func(cr *CheckResponse)) *Status {
-	os.checkers = append(os.checkers, checker{name, checkerFunc})
-	return os
+func (s *Status) AddChecker(name string, checkerFunc func(cr *CheckResponse)) *Status {
+	s.checkers = append(s.checkers, checker{name, checkerFunc})
+	return s
+}
+
+// AddMetrics registers prometheus metrics to be exopsed on the /__/metrics endpoint
+// Adding the same metric twice will result in a panic
+func (s *Status) AddMetrics(cs ...prometheus.Collector) *Status {
+	if s.promRegistry == nil {
```

Comment:
Does this mean it is not possible to use the default registry? 

That means (and maybe this isn't a bad thing) we can't do `prometheus.MustRegister` from anywhere in code and have the metrics appear on the endpoint. We need to have the OpHandler in scope.